From aa6581a7053dd43247d65daa5511fbcc2977731f Mon Sep 17 00:00:00 2001
From: Kurt Kanzenbach <kurt@kmk-computers.de>
Date: Sun, 12 Jul 2020 10:57:08 +0200
Subject: [PATCH] etf: Fix compiler warning/build error in etf plugin
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following warning/error:

|In function ‘snprintf’,
|    inlined from ‘UA_PubSubChannelEthernetETF_open’ at /var/tmp/portage/dev-libs/open62541-1.1/work/open62541-1.1/plugins/ua_pubsub_ethernet_etf.c:189:5,
|    inlined from ‘TransportLayerEthernetETF_addChannel’ at /var/tmp/portage/dev-libs/open62541-1.1/work/open62541-1.1/plugins/ua_pubsub_ethernet_etf.c:563:40:
|/usr/include/bits/stdio2.h:67:10: error: ‘__builtin___snprintf_chk’ specified bound 40 exceeds destination size 16 [-Werror=stringop-overflow=]
|   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
|      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|   68 |        __bos (__s), __fmt, __va_arg_pack ());
|      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|

By using the correct size.

Signed-off-by: Kurt Kanzenbach <kurt@kmk-computers.de>
---
 plugins/ua_pubsub_ethernet_etf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/ua_pubsub_ethernet_etf.c b/plugins/ua_pubsub_ethernet_etf.c
index 416b5857945a..d55eb363e3b5 100644
--- a/plugins/ua_pubsub_ethernet_etf.c
+++ b/plugins/ua_pubsub_ethernet_etf.c
@@ -186,7 +186,7 @@ UA_PubSubChannelEthernetETF_open(const UA_PubSubConnectionConfig *connectionConf
     struct ifreq ifreq;
     memset(&ifreq, 0, sizeof(struct ifreq));
     UA_UInt64 len = UA_MIN(address->networkInterface.length, sizeof(ifreq.ifr_name)-1);
-    UA_snprintf(ifreq.ifr_name, sizeof(struct ifreq),
+    UA_snprintf(ifreq.ifr_name, sizeof(ifreq.ifr_name),
                 "%.*s", (int)len,
                 (char*)address->networkInterface.data);
 
-- 
2.26.2

